scripts/Update-OnETagChanged.ps1

```diff
@@ -27,24 +29,52 @@ function Update-OnETagChanged() {
     $saveResult = $true
   }
   else {
-    $existingInfo = (Get-Content $saveFile -Encoding UTF8 -TotalCount 1) -split '\|'
-
-    if ($existingInfo[0] -ne $etag) {
-      $result = . $OnETagChanged
-      $saveResult = $true
-    }
-    else {
-      $result = . $OnUpdated
-      $result["Version"] = $existingInfo[1]
-      $result["ETAG"] = $existingInfo[0]
-      $saveResult = $false
-    }
+	  if ($kind) {
+		$existingInfo = ( Get-Content $saveFile -Raw | ConvertFrom-Json )
```

Comment:
Don't mix two different formats in this helper, they need to use the same format both when use specify a `kind` and when you don't.

Don't forget about it needing to be backwards compatible as well.